src/App.tsx

```diff
@@ -1,14 +1,41 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
+import { FilteringOptions } from './types/FilteringOptions';
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodosFilteredByStatus, getTodosIncludeSearchQuery } from './helpers/todosFiltering';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filteringOption, setFilteringOption] = useState(FilteringOptions.All);
+  const [searchQuery, setSearchQuery] = useState('');
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [isLoading, setIsLoading] = useState(true);
+  const [hasErrors, setHasErrors] = useState(false);
+
+  useEffect(() => {
+    getTodos()
+      .then(setTodos)
+      .catch(() => setHasErrors(true))
+      .finally(() => setIsLoading(false));
+    // console.log(todos);
+  }, []);
+
+  const todosFilteredByStatus = useMemo(() => {
+    return getTodosFilteredByStatus(todos, filteringOption);
+  }, [filteringOption, isLoading]);
+
+  const todosIncludeSearchQuery = useMemo(() => {
+    return getTodosIncludeSearchQuery(todosFilteredByStatus, searchQuery);
+  }, [filteringOption, searchQuery, isLoading]);
```

Comment:
You should only include 'todosFilteredByStatus' and 'searchQuery' in the dependency array for useMemo as these are the only variables used inside the function.